### DISCUSSION TITLE
Where / how do you actually use Sets and Cases?

### DISCUSSION THREAD

**Discussion by jankapunkt** [openqda/openqda#14] [General]
We want to be full REFI compliant and it covers Sets and Cases, yet we are not 100% sure where/how to implement them in the correct way in terms of QDA. Any input is highly appreciated!

**Comment by kadewe** [comment]
we are using them in a collaborative research seminar. students document their use of AI tools (documents of chats) and their reflection on each chat. they need sets to
(1) organize their individual sessions (with their chats and reflections in one sitting)
(2) organize the material by
students (sets of sets)
(3) organize the material by different session topics (such as coding support or literature search)

so we need to come up (1) with a view for managing sets and at least in the analysis tab to select sets for analysis
(3) for coding and preparing it would he good to hide all sources not contained in an active set. if we allow activating multiple sets (active sets) this allows to combine certain sets (first a logic OR would be good to be extended with AND and XOR

**Reply by kadewe** [reply]
so as a workaround until we have integrated this into OpenQDA, for analysis just export your data as csv, then create another csv with the group or set variables. I am currently creating a collection of python scripts for doing analysis of openqda data to collect feedback on implementation and usefulness. There you can find examples.

**Comment by anouarg88** [comment]
Hi! I was just looking through discussions to see what has been said about _cases_ so far. I agree that _sets_ are useful. At the same time, I think that _cases_ are at least as important. For scenarios where multiple source files are assigned to one case (e.g., multiple interview transcripts to the same participant), it would be nice to have the option to arrange them by case and also to have case-memos. This could be part of the preparation page, where files are imported/created. I think if cases and documents were displayed in a tree structure, for instance, or if users had the option to display only documents belonging to a particular case, this would make navigating through a large corpus of data more convenient. ☮️

**Reply by anouarg88** [reply]
I just realized that it's possible to arrange the documents by file name which means it is already quite easy to sort files by case if you name them accordingly. Also, this would probably make more sense in the Sources page (not the preparation page), after the documents are locked for coding. Sorry for the confusion!

**Comment by kadewe** [comment]
Hi anouarg88, thanks for you insight. Sorting by filename is a workaround, you can even do something like NAME_TYPE_Filename e.g. ANDREW_INTERVIEW_FirstInterview and ANDREW_JOURNALS_110726 to also sort different types of material, but in just prototyping i really tried to understand the REFI way to do it in comparison to other packages.
My understanding of REIF SETs and CASES are:

A **Case** is a unit of analysis. It represents a thing in the world a research is studying — an interviewee, a family, a school, a nursing facility. A Case has 2 properties:
  1. It carries variable values. "Anna: Gender=Female, Age=22, City=Edinburgh" — these describe Anna herself, not any document.
  REFI puts VariableValue elements inside Case, referencing the project-level variable definitions.
  2. It links the evidence about that unit: Anna's interview transcript, a follow-up interview, her diary entries to sort several documents into one case. 
  3. REFI even allows a case to reference parts of documents via selections, such as only Anna's turns in a group-discussion transcript.

  This allows researchers to compare e.g. only women under 30 with women over 50 in a larger collection of cases

  A **Set** is a tool for organizing the researcher's workspace. It is an arbitrary named collection of codes and/or documents that should be in some kind of "folder" for ease of use: "the emotion codes", "the pilot-phase interviews", "everything for chapter 4". Sets do not have variables, they can freely overlap.

 Cases answer "who/what is this data about?", sets answer "which things do I want to look at right now?" 
Sets: filter to only focus on the documents + you need now
Cases: analytical coding object

Right now we have source variables, but these source variables you could understand as variables of a set "ALL SOURCES". We have to talk this through even more to get it right.

**Comment by anouarg88** [comment]
Hey @kadewe, thank you for the file-naming workaround ideas and explaining cases and sets! A case as a unit of analysis is a much less crude understanding than my old idea that a case equals a person. The more I read about it, the more REFI seems like a really cool initiative! It's not just about software compatibility, but about research methodology as well.
